cmudb/env/docker-entrypoint.sh

```diff
@@ -0,0 +1,170 @@
+  echo "listen_addresses = '*'" >> ${AUTO_CONF}
+}
+
+_pg_create_user_and_db() {
+  ${BIN_DIR}/psql -c "create user ${POSTGRES_USER} with login password '${POSTGRES_PASSWORD}'" postgres
+  ${BIN_DIR}/psql -c "create database ${POSTGRES_DB} with owner = '${POSTGRES_USER}'" postgres
+}
+
+_pg_setup_replication() {
+  AUTO_CONF=${PGDATA}/postgresql.auto.conf
+  HBA_CONF=${PGDATA}/pg_hba.conf
+
+  # See PostgreSQL docs for complete description of parameters.
+
+  # wal_level: How much information to ship over.
+  echo "wal_level = replica" >> ${AUTO_CONF}
+  # hot_standby: True to enable connecting and running queries during recovery.
+  echo "hot_standby = on" >> ${AUTO_CONF}
+  # max_wal_senders: Maximum number of concurrent connections to standby/backup clients.
+  echo "max_wal_senders = 10" >> ${AUTO_CONF}
+  # max_replication_slots: Maximum number of replication slots.
+  echo "max_replication_slots = 10" >> ${AUTO_CONF}
+  # hot_standby_feedback: True if standby should tell primary about what queries are currently executing.
+  echo "hot_standby_feedback = on" >> ${AUTO_CONF}
+
+  if [ "${NP_REPLICATION_TYPE}" = "primary" ]; then
+    # ===============================
+    # Enable replication.
+    # ===============================
+
+    # Create replication user.
+    ${BIN_DIR}/psql -c "create user ${NP_REPLICATION_USER} with replication encrypted password '${NP_REPLICATION_PASSWORD}'" postgres
+    # Allow replication user to connect..
+    echo "host replication ${NP_REPLICATION_USER} 0.0.0.0/0 md5" >> ${HBA_CONF}
+    # Reload configuration.
+    ${BIN_DIR}/psql -c "select pg_reload_conf()" postgres
+    # Create replication slot for replica.
+    ${BIN_DIR}/psql -c "select pg_create_physical_replication_slot('replication_slot_replica1')" postgres
+
+    # ===============================
+    # Enable replication stats.
+    # ===============================
+    ${BIN_DIR}/psql -c "grant pg_monitor to ${POSTGRES_USER}" postgres
+  fi
+}
+
+# All the steps required to start up PostgreSQL.
+_pg_start_all() {
+  _pg_initdb              # Initialize a new PostgreSQL cluster.
+  _pg_config              # Write any configuration options required.
+  _pgctl_start            # Start the PostgreSQL cluster.
+  _pg_create_user_and_db  # Create the specified user and database.
+
+  if [ ! -z "${NP_REPLICATION_TYPE}" ]; then
+    _pg_setup_replication
+  fi
+}
+
+# =====================================================================
+# Main logic.
+# =====================================================================
+
+main() {
+  # Only initdb if this is not a replica. The replica will recover from backup.
+  if [ ! "${NP_REPLICATION_TYPE}" = "replica" ]; then
+    # This is a single-node or the primary.
+    _pg_start_all
+    _pg_stop
+    _pg_start
+  else
+    # This is a replica.
+    while true ; do
+      # Note that there is a potential race here where the primary restarts and healthcheck succeeds.
```

Comment:
Should we open an issue to look into this further?

Reply:
Issue #6 opened.